src/apps/frontend/pages/tasks/task-section.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from 'react';
+import toast from 'react-hot-toast';
+
+import {
+  Button,
+  HeadingSmall,
+  LabelLarge,
+  MenuItem,
+  ParagraphSmall,
+  Spinner,
+  VerticalStackLayout,
+} from '../../components';
+import { AsyncError } from '../../types';
+import { ButtonKind, ButtonSize } from '../../types/button';
+import { Task } from '../../types/task';
+
+import TaskModal from './task-modal';
+import useTaskForm from './tasks-form.hook';
+
+interface TaskSectionProps {
+  handleDeleteTask: (taskId: string) => void;
+  isGetTasksLoading: boolean;
+  onError?: (error: AsyncError) => void;
+  tasks: Task[];
+}
+
+const TaskSection: React.FC<TaskSectionProps> = ({
+  handleDeleteTask,
+  isGetTasksLoading,
+  onError,
+  tasks,
+}) => {
+  const [updateTaskModal, setUpdateTaskModal] = useState(false);
+
+  const onSuccess = () => {
+    toast.success('Task has been updated successfully');
+    setUpdateTaskModal(false);
+  };
+
+  const { updateTaskFormik, setFormikFieldValue } = useTaskForm({
+    onError,
+    onSuccess,
+  });
+
+  const handleTaskOperation = (task: Task) => {
+    setUpdateTaskModal(!updateTaskModal);
+    setFormikFieldValue(updateTaskFormik, 'title', task.title);
+    setFormikFieldValue(updateTaskFormik, 'id', task.id);
+    setFormikFieldValue(updateTaskFormik, 'description', task.description);
+  };
+
+  if (isGetTasksLoading) {
+    return (
+      <div className="flex h-96 w-full items-center justify-center">
+        <Spinner />
+      </div>
+    );
+  }
+
+  return (
+    <VerticalStackLayout gap={7}>
+      {tasks.length > 0 && (
+        <HeadingSmall>
+          To Do's ({tasks.length >= 10 ? Math.floor(tasks.length / 10) : '0'}
```

Comment:
`{tasks.length >= 10 ? Math.floor(tasks.length / 10) : '0'` - What is this and why is this needed?

Reply:
![image](https://github.com/jalantechnologies/boilerplate-mern/assets/126680323/9e5af505-1970-412f-b4fd-f79edaea638c)

Logic for the calculation is that tasks are initially counted as 0 at ten's place, then, as tasks increase, the ten's place changes from 0 to 1, and so on.